README.md
pirateBoatIsometricRPGClient
============================

pirateBoatIsometricRPGClient


changelog.txt
[seaking_1.0.0.1 2013-01-19 张春生]
1、新功能：初始化项目

[seaking_1.0.0.1 2013-01-21 张春生]
1、新功能：增加游戏demo

[seaking_1.0.0.1 2013-01-22 张春生]
1、新功能：新建dev分支
2、新功能：增加游戏demo

[seaking_1.0.0.1 2013-01-23 张春生]
1、新功能：增加触摸功能、astar算法

[seaking_1.0.0.1 2013-01-24 张春生]
1、新功能：调整websocket功能

[seaking_1.0.0.1 2013-01-25 张春生]
1、新功能：调整地图动画

[seaking_1.0.0.1 2013-01-25 张春生]
1、新功能：调整跑动动画

[seaking_1.0.0.1 2013-01-26 张春生]
1、新功能：新增45度和直角坐标转换方法

[seaking_1.0.0.1 2013-01-27 张春生]
1、新功能：新增45度和直角坐标转换方法

[seaking_1.0.0.1 2013-01-28 张春生]
1、新功能：调整进度条、增加npc

[seaking_1.0.0.1 2013-01-29 张春生]
1、新功能：调整45度和直角坐标转换方法
2、新功能：调整人物跑动

[seaking_1.0.0.1 2013-01-30 张春生]
1、新功能：调整45度和直角坐标转换方法
2、新功能：调整人物跑动
3、新功能：增加主界面UI

[seaking_1.0.0.1 2013-01-31 张春生]
1、新功能：增加图片资源
2、新功能：增加打包工具
3、新功能：调整文件编码

[seaking_1.0.0.1 2013-02-01 张春生]
1、新功能：调整寻路算法
2、新功能：调整进度条

[seaking_1.0.0.1 2013-02-02 张春生]
1、新功能：调整寻路算法
2、新功能：增加调试信息

[seaking_1.0.0.1 2013-02-03 张春生]
1、新功能：调整寻路算法，增加矩阵变换

[seaking_1.0.0.1 2013-02-04 张春生]
1、新功能：调整寻路算法，在isometric中寻路，调整矩阵变换
2、新功能：调整人物跑动

[seaking_1.0.0.1 2013-02-05 张春生]
1、新功能：调整人物跑动
2、新功能：增加战斗模块

[seaking_1.0.0.1 2013-02-06 张春生]
1、新功能：调整人物跑动

[seaking_1.0.0.1 2013-02-19 張春生]
1、新功能：人物跑動後居中

[seaking_1.0.0.1 2013-02-22 張春生]
1、新功能：增加90度坐標碰撞層尋路功能

[seaking_1.0.0.1 2013-02-23 張春生]
1、新功能：增加90度坐標碰撞層尋路功能
2、新功能：添加遊戲UI

[seaking_1.0.0.1 2013-02-25 張春生]
1、新功能：添加遊戲UI
2、新功能：增加人物屬性信息

[seaking_1.0.0.1 2013-02-26 張春生]
1、重構：重構UI主界面項目代碼
2、重構：增加角色、任務、聊天UI窗口
3、重構：調整文件編碼
4、重構：重構UI樣式代碼
5、重構：重構聊天代碼，增加socket通信

[seaking_1.0.0.1 2013-02-27 張春生]
1、重構：重構UI主界面項目代碼
2、重構：調整尋路為45度角尋路

[seaking_1.0.0.1 2013-02-28 張春生]
1、重構：調整UI主界面項目代碼
2、重構：調整圖片命名

[seaking_1.0.0.1 2013-03-01 張春生]
1、新功能：新增角色窗口
2、重构：重构窗口代码
3、新功能：增加切换场景功能
4、新功能：增加menu按纽

[seaking_1.0.0.1 2013-03-02 張春生]
1、新功能：調整切換場景功能
2、新功能：增加icon圖標
3、新功能：重構UI角色窗口
4、新功能：UI增加menu按鈕點擊效果
5、重構：UI調整左側tab頁圖片資源

[seaking_1.0.0.1 2013-03-03 張春生]
1、新功能：增加戰鬥邏輯

[seaking_1.0.0.1 2013-03-04 張春生]
1、新功能：增加陣型功能
2、重構：重構戰鬥邏輯代碼

[seaking_1.0.0.1 2013-03-05 張春生]
1、重構：重構戰鬥邏輯代碼
2、新功能：UI角色窗口增加reflowPosition方法

[seaking_1.0.0.1 2013-03-06 張春生]
1、新功能：增加戰鬥UI場景
2、新功能：增加圖片資源

[seaking_1.0.0.1 2013-03-07 張春生]
1、重構：重構戰鬥場景

[seaking_1.0.0.1 2013-03-08 張春生]
1、新功能：增加buff效果
2、新功能：增加文字描述
3、新功能：新增新地图
4、重构：UI调整图片资源
5、需求变更：UI调整背包格子位置
6、重构：调整技能窗口

[seaking_1.0.0.1 2013-03-09 張春生]
1、新功能：調整屏幕適配

[seaking_1.0.0.1 2013-03-10 张春生]
1、重构：調整屏幕適配
2、新功能：增加btg.drawFont方法
3、重构：调整websocket服务地址
4、重构：重构战斗场景

[seaking_1.0.0.1 2013-03-11 张春生]
1、重构：重构战斗场景
2、重构：调整添加用户方式
3、重构：调整资源图片

[seaking_1.0.0.67 2013-03-12 张春生]
1、重构：重构战斗场景
2、重构：重构人物跑动
3、重构：调整水上城市

[seaking_1.0.0.67 2013-03-13 张春生]
1、重构：调整战斗场景
2、重构：调整海上城市
3、重构：调整人物属性

[seaking_1.0.0.67 2013-03-14 张春生]
1、重构：UI重构主界面
2、重构：UI调整聊天页面
3、重构：调整按钮样式

[seaking_1.0.0.67 2013-03-16 张春生]
1、重构：调整地图绘制方法

[seaking_1.0.0.67 2013-03-17 张春生]
1、重构：调整地图绘制方法
2、重构：调整打包程序

[seaking_1.0.0.67 2013-03-19 张春生]
1、重构：UI调整主界面
2、重构：UI调整聊天界面
3、重构：重构地砖地图拼接算法

[seaking_1.0.0.67 2013-03-20 张春生]
1、新功能：增加选择角色界面

[seaking_1.0.0.67 2013-03-21 张春生]
1、重构：调整选择角色界面
2、新功能：增加点击效果
3、重构：调整战斗场景

[seaking_1.0.0.67 2013-03-22 张春生]
1、重构：调整点击效果
2、重构：调整加载地图
3、重构：调整目录结构

[seaking_1.0.0.67 2013-03-23 张春生]
1、重构：调整战斗场景
2、重构：新建cocoonjs分支

[seaking_1.0.0.67 2013-03-24 张春生]
1、重构：合并分支并补回之前代码

[seaking_1.0.0.82 2013-03-25 张春生]
1、重构：UI调整主界面

[seaking_1.0.0.82 2013-03-26 张春生]
1、重构：UI调整赌场场景

[seaking_1.0.0.82 2013-03-27 张春生]
1、重构：调整地图层级
2、新功能：UI调整赌场场景
3、新功能：UI增加加载资源

[seaking_1.0.0.82 2013-03-29 张春生]
1、重构：调整websocket重连方式
2、新功能：增加获取任务信息方法

[seaking_1.0.0.82 2013-03-30 张春生]
1、bug：修正其他玩家不能播放站立动画bug

[seaking_1.0.0.82 2013-03-31 张春生]
1、重构：UI重构窗体代码
2、新功能：UI增加人物属性窗口

[seaking_1.0.0.82 2013-04-01 张春生]
1、重构：调整应用打包代码，压缩所有js文件

[seaking_1.0.0.82 2013-04-02 张春生]
1、重构：UI调整角色窗口

[seaking_1.0.0.82 2013-04-04 张春生]
1、重构：增加主角左右动画
1、新功能：增加一个Npc动画

[seaking_1.0.0.82 2013-04-05 张春生]
1、重构：增加水上城市碰撞层
2、重构：更新melonjs框架到0.9.6
3、重构：调整打包代码

[seaking_1.0.0.82 2013-04-06 张春生]
1、新功能：从服务器端获得任务信息
2、重构：调整地图碰撞层
3、重构：新增cocoonjs分支

[seaking_1.0.0.82 2013-04-08 张春生]
1、新功能：任务增加自动寻路功能

[seaking_1.0.0.82 2013-04-09 张春生]
1、新功能：增加任务对话窗口
2、新功能：增加副本资源

[seaking_1.0.0.82 2013-04-11 张春生]
1、新功能：增加副本功能
2、新功能：增加任务功能
3、新功能：增加json格式地图
4、重构：调整新手村地图
5、重构：调整水上城市地图
6、重构：合并分支
7、重构：调整目录结构

--备注分类
1、新功能：
2、Bug:
3、需求变更：
4、重构：
5、merge:

--格式：
[seaking_1.0.0.1 2013-01-19 张春生]
版本号 日期 提交人
版本号：产品名 版本类型 主版本 辅版本 新功能版本 缺陷版本

js/config.js
/**
 * config file
 * author: peter
 * date: 2013-01-28
 */
var g_config = {
	//map
	map: {
		level: [{
			id: 0,
			npcs: [20, 21],
			fightSceneX: 2700,
			fightSceneY: 2000,
			pos: {
				x: 3000,
				y: 1600
			},
			/*mapData: [
				[0,2,3,4,5,0],
				[7,8,9,10,11,0],
				[0,0,15,0,0,0]
			],*/
            polygonData : [
                // 地图外围包含
                [[3018,1914], [2759,2045], [2673,2001], [2717, 1848],
                 [2504,1740], [2547,1717], [2458,1673], [2500, 1650],
                 [2412,1476], [2470,1462], [2499,1475], [2555, 1463],
                 [2800,1451], [2930,1560], [2974,1586], [3232, 1454],
                 [3318,1496], [3100,1607], [3187,1653], [3446, 1522],
                 [3618,1736], [3530,1783], [3660,1848], [3445, 1956],
                 [3360,1911], [3314,1936], [3275,1914], [3144, 1977]],

                // 中部障碍物
                [[2846, 1786], [3061, 1672], [2844, 1563], [2630, 1674]],

                [[3404, 1893], [3320, 1850], [3447, 1784], [3535, 1826]]
            ]
		}, {
			id: 1,
			npcs: [],
			fightSceneX: 2000,
			fightSceneY: 2000,
			pos: {
				x: 3000,
				y: 2000
			},
			/*mapData: [
				[0,2,3,4,5,6,0,0],
				[9,10,11,12,13,14,15,16],
				[17,18,19,20,21,22,23,0],
				[0,0,27,28,29,0,0,0]
			]*/
		}],
		idLevel: [{
			id: 0,
			pos: {
				x: 500,
				y: 1000
			}
		}]
	},

	//character
	character: [{
		id: 0,
		name: "女主角",
		width: 80,
		height: 114,
		"standDown": [0,1,2,3,4,5],
		"standLeft": [72,73,74,75,76,77],
		"standUp": [6,7,8,9,10,11],
		"standRight": [78,79,80,81,82,83],
		"standLeftDown": [30,31,32,33,34,35],
		"standLeftUp": [24,25,26,27,28,29],
		"standRightUp": [12,13,14,15,16,17],
		"standRightDown": [18,19,20,21,22,23],
		"down": [42,43,44,45,46,47],
		"left": [90,91,92,93,94,95],
		"up": [36,37,38,39,40,41],
		"right": [84,85,86,87,88,89],
		"leftDown": [66,67,68,69,70,71],
		"leftUp": [60,61,62,63,64,65],
		"rightUp": [48,49,50,51,52,53],
		"rightDown": [54,55,56,57,58,59]
	},{
		id: 1,
		name: "男主角",
		width: 80,
		height: 114,
        "standDown": [0,1,2,3,4,5],
        "standLeft": [72,73,74,75,76,77],
        "standUp": [6,7,8,9,10,11],
        "standRight": [78,79,80,81,82,83],
        "standLeftDown": [30,31,32,33,34,35],
        "standLeftUp": [24,25,26,27,28,29],
        "standRightUp": [12,13,14,15,16,17],
        "standRightDown": [18,19,20,21,22,23],
        "down": [42,43,44,45,46,47],
        "left": [90,91,92,93,94,95],
        "up": [36,37,38,39,40,41],
        "right": [84,85,86,87,88,89],
        "leftDown": [66,67,68,69,70,71],
        "leftUp": [60,61,62,63,64,65],
        "rightUp": [48,49,50,51,52,53],
        "rightDown": [54,55,56,57,58,59]
	},{
		id: 2
	},{
		id: 3
	},{
		id: 4
	},{
		id: 5
	},{
		id: 6
	},{
		id: 7
	},{
		id: 8
	},{
		id: 9
	},{
		id: 10
	},{
		id: 11
	},{
		id: 12
	},{
		id: 13
	},{
		id: 14
	},{
		id: 15
	},{
		id: 16
	},{
		id: 17
	},{
		id: 18
	},{
		id: 19
	},{
		id: 20,
		name: "npc20",
		width: 110,
		height: 125,
		standLeftDown: [0,1,2,3,4,5],
		shakeLeftDown: [6,7,8,9,10,11,12,13,14,15,16,17,18,19,20],
		standRightDown: [21,22,23,24,25,26],
		shakeRightDown: [27,28,29,30,31,32,33,34,35,36,37,38,39,40,41]
	},{
		id: 21,
		name: "npc21",
		width: 102,
		height: 136,
		standLeftDown: [0,1,2,3,4,5],
		shakeLeftDown: [6,7,8,9,10,11,12,13,14,15,16,17,18,19,20],
		standRightDown: [21,22,23,24,25,26],
		shakeRightDown: [27,28,29,30,31,32,33,34,35,36,37,38,39,40,41]
	},{
		id: 22,
		name: "npc21",
		width: 80,
		height: 117,
		standLeftDown: [0,1,2,3,4,5],
		shakeLeftDown: [6,7,8,9,10,11,12,13,14,15,16,17,18,19,20],
		standRightDown: [21,22,23,24,25,26],
		shakeRightDown: [27,28,29,30,31,32,33,34,35,36,37,38,39,40,41]
	}],

	//pirate boat
	pirateBoat: [{
		id: 0,
		name: "boat1",
		width: 150,
		height: 150,
		"standDown": [0],
		"standLeft": [0],
		"standUp": [0],
		"standRight": [0],
		"standLeftDown": [0],
		"standLeftUp": [0],
		"standRightUp": [0],
		"standRightDown": [0],
		"down": [0],
		"left": [0],
		"up": [0],
		"right": [0],
		"leftDown": [0],
		"leftUp": [0],
		"rightUp": [0],
		"rightDown": [0]
	},{
		id: 1,
		name: "boat2",
		width: 150,
		height: 150,
		"standDown": [0]
	},{
		id: 2,
		name: "boat2",
		width: 150,
		height: 150,
		"standDown": [0]
	},{
		id: 3,
		name: "boat2",
		width: 150,
		height: 150,
		"standDown": [0]
	},{
		id: 4,
		name: "boat2",
		width: 150,
		height: 150,
		"standDown": [0]
	},{
		id: 5,
		name: "boat2",
		width: 150,
		height: 150,
		"standDown": [0]
	},{
		id: 6,
		name: "boat2",
		width: 150,
		height: 150,
		"standDown": [0]
	},{
		id: 7
	},{
		id: 8
	},{
		id: 9
	},{
		id: 10
	},{
		id: 11
	},{
		id: 12
	},{
		id: 13
	},{
		id: 14
	},{
		id: 15
	},{
		id: 16
	},{
		id: 17
	},{
		id: 18
	},{
		id: 19
	},{
		id: 20,
		name: "boat20",
		width: 150,
		height: 150,
		standDown: [0]
	},{
		id: 21
	},{
		id: 22
	}],

	//npc
	npc: {
		"20": {
			id: 20,
			name: "test",
			width: 102,
			height: 136,
			standDown: [0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19]
		}
	},
	
	taskType: ["currentMainTask", "currentBranchTask", "currentDayTask", "currentExerciseTask"],
	symbolType: {
		QGOLD: "questionMark-gold",
		QGRAY: "questionMark-gray",
		EGOLD: "exclamationMark-gold",
		EGRAY: "exclamationMark-gray"
	},

	//main task
	mainTask: [{
		id: 0,
		name: "",
		getItems: [{
			id: 0,
			num: 10
		}]
	}],

	//day task
	dayTask: [{
		id: 0,
		name: "",
		getItems: [{
			id: 0,
			num: 10
		}]
	}],

	//items
	items: [{
		id: 0,
		name: "",
		type: 1
	}],
	
	//skills
	skills: [{
		id: 0,
		name: ""
	}],

	//formation
	formation: {
		bg: {
			left: 0,
			top: 219,
			width: 476,
			height: 350,
			marginLeft: 8,
			positionWidth: 90,
			positionHeight: 60
		},
		positions: {
			mainPlayer: [{
				x: 234,
				y: 167
			}, {
				x: 266,
				y: 39
			}, {
				x: 408,
				y: 114
			}, {
				x: 387,
				y: 246
			}, {
				x: 200,
				y: 304
			}, {
				x: 48,
				y: 218
			}, {
				x: 90,
				y: 87
			}],
			enemy: [{
				x: 243,
				y: 167
			}, {
				x: 210,
				y: 39
			}, {
				x: 68,
				y: 114
			}, {
				x: 89,
				y: 246
			}, {
				x: 277,
				y: 304
			}, {
				x: 428,
				y: 218
			}, {
				x: 386,
				y: 87
			}]
		}
	}
};
g_config.pirateBoat[300] = {
	id: 300,
	name: "boat1",
	width: 150,
	height: 150,
	"standDown": [0],
	"standLeft": [0],
	"standUp": [0],
	"standRight": [0],
	"standLeftDown": [0],
	"standLeftUp": [0],
	"standRightUp": [0],
	"standRightDown": [0],
	"down": [0],
	"left": [0],
	"up": [0],
	"right": [0],
	"leftDown": [0],
	"leftUp": [0],
	"rightUp": [0],
	"rightDown": [0]
};